packages/vite/src/node/utils.ts

```diff
@@ -511,6 +511,11 @@ export function generateCodeFrame(
     end !== undefined ? posToNumber(source, end) : start,
     source.length,
   )
+  const lastPosLine =
+    end !== undefined
+      ? numberToPos(source, end).line
+      : numberToPos(source, start).line + range
+  const lineNumberWidth = Math.max(3, Math.floor(Math.log10(lastPosLine)) + 1)
```

Comment:
It took me a while to understand why its doing a log10 here. I think the previous `String().length` was easier to understand. Is there a reason you switched to this?